### PR TITLE
webcat: init at unstable-2021-09-06

### PR DESCRIPTION
**Pull request by montag451**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/webcat/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    rev = version;
+    sha256 = "15c62sjr15l5hwkvc4xarfn76341wi16pjv9qbr1agaz1vqgr6rd";
+  };
+
+  vendorSha256 = "1apnra58mqrazbq53f0qlqnyyhjdvvdz995yridxva0fxmwpwcjy";
+
+  meta = with lib; {
+    homepage = "https://rumpelsepp.org/blog/ssh-through-websocket/";
+    description = "The lightweight swiss army knife for websockets";
+    licence = lib.gpl3;
```
Comment (Pacman99) [inline pkgs/tools/misc/webcat/default.nix:18]:
```suggestion
    licence = gpl3;
```
I believe you also have to specify `gpl3Only` or `gpl3Plus`, I think this program is `gpl3Plus`.

File: pkgs/tools/misc/webcat/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    maintainers = [ maintainers.montag451 ];
```
Comment (Pacman99) [inline pkgs/tools/misc/webcat/default.nix:19]:
```suggestion
    maintainers = with maintainers; [ montag451 ];
```

File: pkgs/tools/misc/webcat/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{ lib, buildGoModule, fetchgit }:
+
+buildGoModule rec {
+  pname = "webcat";
+  version = "57a65558f0affac0b2f8f4831c52964eb9ad5386";
+
+  src = fetchgit {
+    url = "https://codeberg.org/rumpelsepp/webcat.git";
+    rev = version;
```
Comment (Pacman99) [inline pkgs/tools/misc/webcat/default.nix:9]:
```suggestion
    rev = "57a65558f0affac0b2f8f4831c52964eb9ad5386";
```

**Review by Pacman99 — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/webcat/default.nix [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{ lib, buildGoModule, fetchgit }:
+
+buildGoModule rec {
+  pname = "webcat";
+  version = "57a65558f0affac0b2f8f4831c52964eb9ad5386";
```
[the remark this answers is not in the document]
Comment (montag451) [reply pkgs/tools/misc/webcat/default.nix]: Done in c786f97

File: pkgs/tools/misc/webcat/default.nix (hunk 1 above)
Comment (montag451) [reply pkgs/tools/misc/webcat/default.nix]: Done in c786f97

File: pkgs/tools/misc/webcat/default.nix (hunk 3 above)
Comment (montag451) [reply pkgs/tools/misc/webcat/default.nix]: Done in c786f97

File: pkgs/tools/misc/webcat/default.nix (hunk 2 above)
Comment (montag451) [reply pkgs/tools/misc/webcat/default.nix]: Done in c786f97

**Comment by montag451** [comment]
Hi @Pacman99,

Thx for the review, I have made all the changes you have suggested in c786f97. Hope it's OK for you now :smiley:

**Review by Pacman99 — APPROVED** [review]
LGTM

File: pkgs/tools/misc/webcat/default.nix [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+buildGoModule rec {
+  pname = "webcat";
+  version = "unstable-20210906";
+
+  src = fetchgit {
+    url = "https://codeberg.org/rumpelsepp/webcat.git";
+    rev = "57a65558f0affac0b2f8f4831c52964eb9ad5386";
+    sha256 = "15c62sjr15l5hwkvc4xarfn76341wi16pjv9qbr1agaz1vqgr6rd";
+  };
```
Comment (figsoda) [inline pkgs/tools/misc/webcat/default.nix:11]: `fetchFromGitea` might be better here

File: pkgs/tools/misc/webcat/default.nix [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+  vendorSha256 = "1apnra58mqrazbq53f0qlqnyyhjdvvdz995yridxva0fxmwpwcjy";
+
+  meta = with lib; {
+    homepage = "https://rumpelsepp.org/blog/ssh-through-websocket/";
+    description = "The lightweight swiss army knife for websockets";
+    license = licenses.gpl3Plus;
```
Comment (figsoda) [inline pkgs/tools/misc/webcat/default.nix:18]: are you sure this is gpl3Plus? I can't find any mentions of "any later version"

File: pkgs/tools/misc/webcat/default.nix [hunk 7]
```diff
@@ -0,0 +1,21 @@
+{ lib, buildGoModule, fetchgit }:
+
+buildGoModule rec {
+  pname = "webcat";
+  version = "unstable-20210906";
```
Comment (figsoda) [inline pkgs/tools/misc/webcat/default.nix:5]:
```suggestion
  version = "unstable-2021-09-06";
```

**Review by figsoda — CHANGES_REQUESTED** [review]
upstream includes a man page that can be generated with asciidoc, it would be nice to have in this package too

File: pkgs/tools/misc/webcat/default.nix (hunk 6 above)
Comment (montag451) [reply pkgs/tools/misc/webcat/default.nix:18]: To be honest, I don't know. I've followed the advice of @Pacman99 but you may be right. I will switch back to `gpl3Only`

File: pkgs/tools/misc/webcat/default.nix (hunk 5 above)
Comment (montag451) [reply pkgs/tools/misc/webcat/default.nix:11]: You are right, I will use `fetchFromGitea`

File: pkgs/tools/misc/webcat/default.nix (hunk 6 above)
Comment (Pacman99) [reply pkgs/tools/misc/webcat/default.nix:18]: Yeah I'm not sure, I didn't see any text at the top of any of the source files. I'm not sure what we should default to if its not specified.

Comment (figsoda) [reply pkgs/tools/misc/webcat/default.nix:18]: the default would be gpl3Only

File: pkgs/tools/misc/webcat/default.nix (hunk 5 above)
Comment (montag451) [reply pkgs/tools/misc/webcat/default.nix]: Done

**Comment by montag451** [comment]
@figsoda I've made all the changes requested! Thx for your review :+1:

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 140202` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>webcat</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]

**Pull request merged**